### PR TITLE
DCP-3398 : Ena File Downloader removing starting and trailing spaces …

### PR DESCRIPTION
**Pull request by swatisuman1990**
…from input parameters

### CODE REVIEW

File: src/main/java/uk/ac/ebi/ena/app/MainRunner.java [hunk 1]
```diff
@@ -163,5 +165,35 @@ public void run(String... args) {
         }
     }
 
+    private void trimInputs(String formatStr, String protocolStr, String downloadLocation,
+                            String asperaLocation, String accessions, String userName, String password,
+                            String emailId) {
+        if (formatStr != null) {
```
Comment (suranjayathilaka) [inline src/main/java/uk/ac/ebi/ena/app/MainRunner.java:171]: use StringUtils.trim() ?

File: src/main/java/uk/ac/ebi/ena/app/MainRunner.java [hunk 2]
```diff
@@ -103,6 +103,8 @@ public void run(String... args) {
 
             if (args.length >= 5) {
                 try {
+                    trimInputs(formatStr, protocolStr, downloadLocation, asperaLocation, accessions, userName, password,
+                            emailId);
                     DownloadFormatEnum format = DownloadFormatEnum.valueOf(formatStr);
```
Comment (suranjayathilaka) [inline src/main/java/uk/ac/ebi/ena/app/MainRunner.java:108]: for params that are only used once can do trim where it's used? e.g. formatStr

File: src/main/java/uk/ac/ebi/ena/app/menu/services/MenuService.java [hunk 3]
```diff
@@ -150,6 +151,7 @@ private DownloadJob a2GetAccessionListFromCommaSeparated(AccessionsEntryMethodEn
         } else if (inputValues.equalsIgnoreCase("b")) { // back
             aBuildAccessionEntryMenu(authenticationDetail);
         } else {
+            inputValues = inputValues.trim();
```
Comment (suranjayathilaka) [inline src/main/java/uk/ac/ebi/ena/app/menu/services/MenuService.java:154]: do the trim where you get this value

File: src/main/java/uk/ac/ebi/ena/app/menu/services/MenuService.java [hunk 4]
```diff
@@ -368,6 +374,7 @@ private String eRequestEmailId(DownloadFormatEnum format, String location, Downl
         if (input.equalsIgnoreCase("b")) { // back
             dRequestProtocolSelection(format, location, downloadJob, authenticationDetail);
         } else if (StringUtils.isNotEmpty(input)) {
+            input = input.trim();
```
Comment (suranjayathilaka) [inline src/main/java/uk/ac/ebi/ena/app/menu/services/MenuService.java:377]: trim at the source

File: src/main/java/uk/ac/ebi/ena/app/MainRunner.java [hunk 5]
```diff
@@ -103,7 +103,10 @@ public void run(String... args) {
 
             if (args.length >= 5) {
                 try {
+                    trimInputs(downloadLocation, asperaLocation, accessions, userName);
+                    formatStr = StringUtils.trim(formatStr);
```
Comment (suranjayathilaka) [inline src/main/java/uk/ac/ebi/ena/app/MainRunner.java:107]: can reduce a line of code by doing the trim when you pass to enum?
Also, should we not do blank checks on these params? what happens if they are null?

File: src/main/java/uk/ac/ebi/ena/app/MainRunner.java [hunk 6]
```diff
@@ -103,7 +103,10 @@ public void run(String... args) {
...
                     DownloadFormatEnum format = DownloadFormatEnum.valueOf(formatStr);
+                    protocolStr = StringUtils.trim(protocolStr);
```
Comment (suranjayathilaka) [inline src/main/java/uk/ac/ebi/ena/app/MainRunner.java:109]: can give FTP as default?
something like @Value("${protocol:#{FTP}}") ?

File: src/main/java/uk/ac/ebi/ena/app/menu/services/MenuService.java [hunk 7]
```diff
@@ -229,7 +232,7 @@ public String requestForDataHubPassword() {
         String input = scannerUtils.getNextString();
         MenuUtils.printEmptyline();
         if (StringUtils.isNotEmpty(input)) {
```
Comment (suranjayathilaka) [inline src/main/java/uk/ac/ebi/ena/app/menu/services/MenuService.java:234]: can replace these 4 lines with a single line

File: src/main/java/uk/ac/ebi/ena/app/menu/services/MenuService.java [hunk 8]
```diff
@@ -364,6 +370,7 @@ private String eRequestEmailId(DownloadFormatEnum format, String location, Downl
         MenuUtils.printBackMessage();
 
         String input = scannerUtils.getNextString();
```
Comment (suranjayathilaka) [inline src/main/java/uk/ac/ebi/ena/app/menu/services/MenuService.java:372]: why not wrap scannerUtils.getNextString() in trim directly? this applies to all places where you do scannerutils

File: src/main/java/uk/ac/ebi/ena/app/MainRunner.java (hunk 5 above)
Comment (swatisuman1990) [reply src/main/java/uk/ac/ebi/ena/app/MainRunner.java:107]: StringUtils.trim  handles for NPE so no need for null check.

File: src/main/java/uk/ac/ebi/ena/app/menu/services/MenuService.java [hunk 9]
```diff
@@ -457,7 +455,7 @@ private String d1RequestAsperaConnectOption(DownloadFormatEnum format, String lo
         System.out.println("***** " + ASPERA_PATH_MSG);
         CommonUtils.printSeparatorLine();
         MenuUtils.printBackMessage();
-        String input = scannerUtils.getNextString();
+        String input = StringUtils.trim(scannerUtils.getNextString());
```
Comment (suranjayathilaka) [inline src/main/java/uk/ac/ebi/ena/app/menu/services/MenuService.java:458]: can put the trim inside getNextString so it covers everything?

File: src/main/java/uk/ac/ebi/ena/app/constants/Constants.java [hunk 10]
```diff
@@ -45,5 +45,7 @@ public class Constants {
     public static final String URLENCODED = "application/x-www-form-urlencoded";
     public static final String APPLICATION_JSON = "application/json";
 
+    public static final String FTP_PROTOCOL = "FTP";
```
Comment (suranjayathilaka) [inline src/main/java/uk/ac/ebi/ena/app/constants/Constants.java:48]: why add a const when we have an enum?

File: src/main/java/uk/ac/ebi/ena/app/MainRunner.java [hunk 11]
```diff
@@ -163,5 +167,25 @@ public void run(String... args) {
...
+        this.accessions = StringUtils.trim(accessions);
+        this.userName = StringUtils.trim(userName);
+    }
+
+    private void validateInputs(String accessions, String formatStr, String downloadLocation, String protocolStr,
+                                String asperaLocation) {
+        if (StringUtils.isEmpty(accessions)) {
+            throw new IllegalArgumentException("Please provide comma separated list of accessions or file path to the accession list");
+        } else if (StringUtils.isEmpty(formatStr)) {
+            throw new IllegalArgumentException("Please provide the format for the download (`eg : READS_FASTQ,READS_SUBMITTED,ANALYSIS_SUBMITTED,ANALYSIS_GENERATED`)");
```
Comment (suranjayathilaka) [inline src/main/java/uk/ac/ebi/ena/app/MainRunner.java:182]: what if format is not empty but invalid?

Comment (suranjayathilaka) [reply src/main/java/uk/ac/ebi/ena/app/MainRunner.java:182]: same for protocol

**Review by suranjayathilaka — APPROVED** [review]

**Pull request merged**